src/CMakeLists.txt

```diff
@@ -739,35 +756,35 @@ install(FILES ${CMAKE_BINARY_DIR}/src/tinyspline.pc
 
 # C++ library
 if(TINYSPLINE_ENABLE_CXX)
-	add_library(tinysplinecxx_shared SHARED ${TINYSPLINE_CXX_SOURCE_FILES})
-	set_target_properties(tinysplinecxx_shared PROPERTIES
+	add_library(tinysplinecxx ${TINYSPLINE_CXX_SOURCE_FILES})
+    if(NOT BUILD_SHARED_LIBS)
+        target_compile_definitions(tinysplinecxx PUBLIC TINYSPLINECXX_STATIC)
+    endif()
+	set_target_properties(tinysplinecxx PROPERTIES
 		OUTPUT_NAME "${TINYSPLINE_CXX_LIBRARIES_OUTPUT_NAME}"
 		DEBUG_POSTFIX "${TINYSPLINE_DEBUG_POSTFIX}"
 		COMPILE_FLAGS "${TINYSPLINE_LIBRARY_CXX_FLAGS}")
-	install(TARGETS tinysplinecxx_shared
+	install(TARGETS tinysplinecxx EXPORT tinysplinecxx
 		LIBRARY DESTINATION ${TINYSPLINE_INSTALL_LIBRARY_DIR}
 		ARCHIVE DESTINATION ${TINYSPLINE_INSTALL_LIBRARY_DIR}
-		RUNTIME DESTINATION ${TINYSPLINE_INSTALL_LIBRARY_DIR})
-	target_include_directories(tinysplinecxx_shared
-			PUBLIC ${TINYSPLINE_CXX_INCLUDE_DIR})
-	add_library(tinysplinecxx_static STATIC ${TINYSPLINE_CXX_SOURCE_FILES})
-	set_target_properties(tinysplinecxx_static PROPERTIES
-		OUTPUT_NAME "${TINYSPLINE_CXX_LIBRARIES_OUTPUT_NAME}"
-		DEBUG_POSTFIX "${TINYSPLINE_DEBUG_POSTFIX}"
-		PREFIX "lib"
-		COMPILE_FLAGS "${TINYSPLINE_LIBRARY_CXX_FLAGS}")
-	install(TARGETS tinysplinecxx_static
-		LIBRARY DESTINATION ${TINYSPLINE_INSTALL_LIBRARY_DIR}
-		ARCHIVE DESTINATION ${TINYSPLINE_INSTALL_LIBRARY_DIR})
-	target_include_directories(tinysplinecxx_static
-		PUBLIC ${TINYSPLINE_CXX_INCLUDE_DIR})
+		RUNTIME DESTINATION ${TINYSPLINE_INSTALL_BINARY_DIR})
+    install(EXPORT tinysplinecxx
+        DESTINATION "${TINYSPLINE_INSTALL_LIBRARY_DIR}/${TINYSPLINE_INSTALL_CMAKE_CONFIG_DIR}/tinysplinecxx"
+        NAMESPACE "tinysplinecxx::"
+        FILE tinysplinecxx-targets.cmake)
+	target_include_directories(tinysplinecxx
+			PUBLIC
+			    $<BUILD_INTERFACE:${TINYSPLINE_CXX_INCLUDE_DIR}>
+			    $<INSTALL_INTERFACE:${TINYSPLICE_INSTALL_INCLUDE_DIR}>)
+    target_link_libraries(tinysplinecxx PRIVATE ${TINYSPLINE_CXX_STATIC_LIBRARIES})
```

Comment:
See comment: `target_link_libraries(tinyspline ${TINYSPLINE_C_STATIC_LIBRARIES})`.

Also, why are you using `PRIVATE` here but not in `target_link_libraries(tinyspline ${TINYSPLINE_C_STATIC_LIBRARIES})`?

Reply:
See https://github.com/msteinbeck/tinyspline/pull/167#discussion_r502619328